astacus/node/sqlite_snapshot.py

```diff
@@ -89,7 +89,16 @@ def _con(self) -> sqlite3.Connection:
             self.db.unlink()
         else:
             self.db.parent.mkdir(parents=True, exist_ok=True)
-        con = sqlite3.connect(self.db, isolation_level=None, check_same_thread=False)
+        # Python 3.12+ fetch result are not consistent in multi-threading application
```

Comment:
Changed :+1: 
Based on https://sqlite.org/threadsafe.html the default value during compile-time is serialized (3)
and the [sqlite3.threadsafety](https://docs.python.org/3/library/sqlite3.html#sqlite3.threadsafety) is set based on the default threading mode the underlying SQLite library is compiled with. However I will check this.

Reply:
We don't build sqlite3 - fedora does. The default threadsafe setting for upstream sqlite doesn't matter to us - what happens if our fedora package maintainer accidentally or intentionally changes this?
 
the point is, in astacus we use sqlite3 in a way that REQUIRES threadsafety to be 3. Otherwise everything will collapse; it's an implicit requirement, we should make it explicit. Maybe a unit test would work as well - but for those things I think that a runtime check asserting the expecting state of the environment is the way to go.

Reply:
- added assertion, manually tested: when we assert 1(single thread) some test will fail
- added comment, with reference to the issue